### ISSUE TITLE
Catchpoint Stuck on Phase 1

### ISSUE THREAD

**Issue by Blackglade** [algorand/indexer#1155]
### Subject of the issue

I'm running an indexer mainnet instance and I recently upgraded from 2.11 => 2.13. I went through the procedures for updating to the last catchpoint but it seems to have gotten stuck after processing the accounts. Here are logs:

```
Jul 29 21:32:04 ip-10-0-1-62 algorand-indexer[10434]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T21:32:04Z"}
Jul 29 21:32:09 ip-10-0-1-62 algorand-indexer[10434]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T21:32:09Z"}
Jul 29 21:32:14 ip-10-0-1-62 algorand-indexer[10434]: {"event":"ConnectedOut","file":"wsNetwork.go","function":"github.com/algorand/go-algorand/network.(*WebsocketNetwork).tryConnect","level":"info","line":2094,"local":"","msg":"Made outgoing connection to peer relay-mumbai-mai
Jul 29 21:32:14 ip-10-0-1-62 algorand-indexer[10434]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T21:32:14Z"}
Jul 29 21:32:19 ip-10-0-1-62 algorand-indexer[10434]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T21:32:19Z"}
Jul 29 21:32:24 ip-10-0-1-62 algorand-indexer[10434]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T21:32:24Z"}
Jul 29 21:32:29 ip-10-0-1-62 algorand-indexer[10434]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T21:32:29Z"}
Jul 29 21:32:34 ip-10-0-1-62 algorand-indexer[10434]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T21:32:34Z"}
Jul 29 21:32:39 ip-10-0-1-62 algorand-indexer[10434]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T21:32:39Z"}
```

It's all normal except the WebSocket connection issue. One thing to note is that my indexer instance is in a VPC on AWS. The firewall rules allow for only incoming SSH connections, but no outgoing connections. It's been an hour at this point with the indexer stuck in this state.

### Your environment

12885426177
3.8.1.stable [rel/stable] (commit #73615e0b)
go-algorand is licensed with AGPLv3.0
source code available at https://github.com/algorand/go-algorand
admin@ip-10-0-1-62:/mainnet$

**Comment by Blackglade** [comment]
Other log errors I get:
```
Jul 29 21:41:23 ip-10-0-1-62 algorand-indexer[11463]: {"event":"Disconnected","file":"wsNetwork.go","function":"github.com/algorand/go-algorand/network.(*WebsocketNetwork).removePeer","level":"info","line":2185,"local":"","msg":"Peer r-no.algorand-mainnet.network:4160 disconnect
Jul 29 21:41:23 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T21:41:23Z"}
Jul 29 21:41:23 ip-10-0-1-62 algorand-indexer[11463]: {"event":"ConnectedOut","file":"wsNetwork.go","function":"github.com/algorand/go-algorand/network.(*WebsocketNetwork).tryConnect","level":"info","line":2094,"local":"","msg":"Made outgoing connection to peer r-y4.algorand-mai
```

**Comment by Blackglade** [comment]
```
Jul 29 21:57:38 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T21:57:38Z"}
Jul 29 21:57:43 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T21:57:43Z"}
Jul 29 21:57:45 ip-10-0-1-62 algorand-indexer[11463]: {"callee":"github.com/algorand/go-algorand/ledger.(*CatchpointCatchupAccessorImpl).BuildMerkleTrie.func1","caller":"/go/src/github.com/algorand/indexer/third_party/go-algorand/ledger/catchupaccessor.go:463","file":"dbutil.go","function":"github.com/algorand/go-algorand/util/db.(*Accessor).atomic","level":"warning","line":304,"msg":"dbatomic: tx surpassed expected deadline by 2m39.677952187s","readonly":false,"time":"2022-07-29T21:57:45Z"}
Jul 29 21:57:45 ip-10-0-1-62 algorand-indexer[11463]: {"file":"catchpointService.go","function":"github.com/algorand/go-algorand/catchup.(*CatchpointCatchupService).processStageLedgerDownload","level":"warning","line":324,"msg":"unable to download ledger : database or disk is full","time":"2022-07-29T21:57:45Z"}
Jul 29 21:57:48 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T21:57:48Z"}
```

**Comment by Blackglade** [comment]
Yep okay going to downgrade my indexer at the moment, but it waits for about 8 minutes and then just restarts...

```
}
Jul 29 22:36:23 ip-10-0-1-62 algorand-indexer[11463]: {"event":"Disconnected","file":"wsNetwork.go","function":"github.com/algorand/go-algorand/network.(*WebsocketNetwork).removePeer","level":"info","line":2185,"local":"","messageDelay":1181334173,"msg":"Peer r-es.algorand-mainnet.network:4160 disconnected: LeastPerformingPeer","remote":"r-es.algorand-mainnet.network:4160","time":"2022-07-29T22:36:23Z"}
Jul 29 22:36:23 ip-10-0-1-62 algorand-indexer[11463]: {"event":"ConnectedOut","file":"wsNetwork.go","function":"github.com/algorand/go-algorand/network.(*WebsocketNetwork).tryConnect","level":"info","line":2094,"local":"","msg":"Made outgoing connection to peer r-o9.algorand-mainnet.network:4160","remote":"r-o9.algorand-mainnet.network:4160","time":"2022-07-29T22:36:23Z"}
Jul 29 22:36:23 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:36:23Z"}
Jul 29 22:36:28 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:36:28Z"}
Jul 29 22:36:33 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:36:33Z"}
Jul 29 22:36:38 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:36:38Z"}
Jul 29 22:36:43 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:36:43Z"}
Jul 29 22:36:48 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:36:48Z"}
Jul 29 22:36:53 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:36:53Z"}
Jul 29 22:36:58 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:36:58Z"}
Jul 29 22:37:03 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:37:03Z"}
Jul 29 22:37:08 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:37:08Z"}
Jul 29 22:37:13 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:37:13Z"}
Jul 29 22:37:18 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:37:18Z"}
Jul 29 22:37:23 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:37:23Z"}
Jul 29 22:37:28 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:37:28Z"}
Jul 29 22:37:33 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14233616 / 14233616","time":"2022-07-29T22:37:33Z"}
Jul 29 22:37:36 ip-10-0-1-62 algorand-indexer[11463]: {"callee":"github.com/algorand/go-algorand/ledger.(*CatchpointCatchupAccessorImpl).ResetStagingBalances.func1","caller":"/go/src/github.com/algorand/indexer/third_party/go-algorand/ledger/catchupaccessor.go:191","file":"dbutil.go","function":"github.com/algorand/go-algorand/util/db.(*Accessor).atomic","level":"warning","line":304,"msg":"dbatomic: tx surpassed expected deadline by 8m49.061025022s","readonly":false,"time":"2022-07-29T22:37:36Z"}
Jul 29 22:37:38 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 21504 / 14233616","time":"2022-07-29T22:37:38Z"}
Jul 29 22:37:43 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 114688 / 14233616","time":"2022-07-29T22:37:43Z"}
Jul 29 22:37:48 ip-10-0-1-62 algorand-indexer[11463]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 205824 / 14233616","time":"2022-07-29T22:37:48Z"}
```

@winder

**Comment by winder** [comment]
@Blackglade thanks for the logs. I don't recall seeing anything like this during testing, so I'll need ask around next week.

**Comment by winder** [comment]
> It's all normal except the WebSocket connection issue. One thing to note is that my indexer instance is in a VPC on AWS. The firewall rules allow for only incoming SSH connections, but no outgoing connections. It's been an hour at this point with the indexer stuck in this state.

Could you tell me a little more about the VPC configuration, and the firewall rules?

From a network perspective there are two main operations:
1) download the ledger / catchpoint file (with account balances).
2) download 1000 blocks.

From the logs you shared, it seems clear that the ledger downloads properly, then it gets stuck downloading blocks. I'm going to investigate how network connections may differ in how the ledger/blocks are fetched.

**Comment by Blackglade** [comment]
Sure! So my VPC has 3 private subnets and 1 public subnet. Most of the traffic in the private subnet is connected to the public internet via a NAT Gateway with an Elastic IP Address. My route table defines a catch-all allowing any traffic connecting to outside the VPC defined CIDR block to be funneled through the NAT Gateway.

The EC2 instance in question holding the indexer instance is in a private subnet with the following security group settings:

Inbound:
Type | Protocol | Port Range | Source
-- | -- | -- | --
SSH | TCP | 22 | 0.0.0.0/0

Outbound:
Type | Protocol | Port Range | Source
-- | -- | -- | --
All Traffic | All | All | 0.0.0.0/0

**Comment by onetechnical** [comment]
> ```
> Jul 29 21:57:45 ip-10-0-1-62 algorand-indexer[11463]: {"file":"catchpointService.go","function":"github.com/algorand/go-algorand/catchup.(*CatchpointCatchupService).processStageLedgerDownload","level":"warning","line":324,"msg":"unable to download ledger : database or disk is full","time":"2022-07-29T21:57:45Z"}

Just saw this line; did it run out of disk?

**Comment by Blackglade** [comment]
> Just saw this line; did it run out of disk?

No idea why it printed this out. I ran `df -H` and my local volume / connected EBS disks had plenty of space

**Comment by winder** [comment]
Is your Indexer data directory pointed at that EBS disk?

**Comment by Blackglade** [comment]
Yes

**Comment by winder** [comment]
@Blackglade thanks for bearing with me. I ran some tests and closer inspection of your error messages, I have a few things to share:

# Unusual messages
* Many of the messages you saw are normal:
    * github.com/algorand/go-algorand/network.(*WebsocketNetwork).tryConnect
    * github.com/algorand/go-algorand/network.(*WebsocketNetwork).removePeer
    * github.com/algorand/go-algorand/ledger.(*CatchpointCatchupAccessorImpl).BuildMerkleTrie.func1
        * I'm not sure if this one is normal, but I saw it also. Maybe needs to be suppressed.
* This one is not normal, and is a bit alarming:
    * github.com/algorand/go-algorand/catchup.(*CatchpointCatchupService).processStageLedgerDownload
        * "unable to download ledger : database or disk is full"

# Processed accounts resetting
* github.com/algorand/go-algorand/ledger.(*CatchpointCatchupAccessorImpl).ResetStagingBalances.func1
    * dbatomic: tx surpassed expected deadline by 8m49.061025022s

I was able to reproduce this using an EBS drive once it started throttling the IOPS.

# Followup question / recommendation

What type of EBS drive are you using, and how many IOPS has it been provisioned with?

New recommendation: deployment should use `NVMe` drives.

This is based on testing with `standard (magnetic)` / `gp2` / `io1` / `io2` / `NVMe` and matches the recommendation for algod. It should have been the recommendation for the new version of Indexer from the beginning, so I'm really sorry to have put you through this. Thanks again for the detailed logs that made it very clear that there was a problem.

Furthermore, for this testing I also put together a utility to make it easier to test this process. I hope to make it available in a future release to assist with debugging hardware configurations in the future.

**Comment by Blackglade** [comment]
Thanks @winder, this is some great information to go off of. 

So I had only tested this on my mainnet node 2TB EBS volume which is a `gp3` drive having a throughput of 600 MB/s and 9000 Provisioned IOPS. I believe this is an NVMe volume as well. Here is a screenshot of the metrics from the last week:

![Screen Shot 2022-08-02 at 12 01 37 PM](https://user-images.githubusercontent.com/6363077/182421423-7962c9d7-de11-4060-a6d5-39d677c7ce71.png)

The spike was most definitely during the initial catchpoint processing. Maybe it makes sense for me to switch over to `io2` or potentially just temporarily increase the IOPS for my `gp3` volume?

**Comment by winder** [comment]
If you are using a volume with 9000 Provisioned IOPS this may have been a fluke.

One of my tests used a 32 GB `io2` volume on a `t2.micro` instance and the process completed after 62 minutes.

**Comment by Blackglade** [comment]
Hmmm so I deleted all the indexer generated files (.sqlite) and reran. Same issue:

```
Aug 02 18:02:18 ip-10-0-1-62 algorand-indexer[28111]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14246400 / 14263996","time":"2022-08-02T18:02:18Z"}
Aug 02 18:02:23 ip-10-0-1-62 algorand-indexer[28111]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14255616 / 14263996","time":"2022-08-02T18:02:23Z"}
Aug 02 18:02:28 ip-10-0-1-62 algorand-indexer[28111]: {"file":"ledgerFetcher.go","function":"github.com/algorand/go-algorand/catchup.(*ledgerFetcher).getPeerLedger.func1","level":"info","line":145,"msg":"writing balances to disk took 1110 seconds, writing creatables to disk took 140 seconds, writing hashes to disk took 77 seconds, total duration is 2106 seconds","time":"2022-08-02T18:02:28Z"}
Aug 02 18:02:28 ip-10-0-1-62 algorand-indexer[28111]: {"file":"catchpointService.go","function":"github.com/algorand/go-algorand/catchup.(*CatchpointCatchupService).processStageLedgerDownload","level":"info","line":300,"msg":"ledger downloaded in 2135 seconds","time":"2022-08-02T18:02:28Z"}
Aug 02 18:02:28 ip-10-0-1-62 algorand-indexer[28111]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14263996 / 14263996","time":"2022-08-02T18:02:28Z"}
Aug 02 18:02:33 ip-10-0-1-62 algorand-indexer[28111]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14263996 / 14263996","time":"2022-08-02T18:02:33Z"}
Aug 02 18:02:38 ip-10-0-1-62 algorand-indexer[28111]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14263996 / 14263996","time":"2022-08-02T18:02:38Z"}
Aug 02 18:02:43 ip-10-0-1-62 algorand-indexer[28111]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14263996 / 14263996","time":"2022-08-02T18:02:43Z"}
Aug 02 18:02:48 ip-10-0-1-62 algorand-indexer[28111]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14263996 / 14263996","time":"2022-08-02T18:02:48Z"}
Aug 02 18:02:53 ip-10-0-1-62 algorand-indexer[28111]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14263996 / 14263996","time":"2022-08-02T18:02:53Z"}
Aug 02 18:02:58 ip-10-0-1-62 algorand-indexer[28111]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14263996 / 14263996","time":"2022-08-02T18:02:58Z"}
Aug 02 18:03:01 ip-10-0-1-62 algorand-indexer[28111]: {"callee":"github.com/algorand/go-algorand/ledger.(*CatchpointCatchupAccessorImpl).BuildMerkleTrie.func1","caller":"/go/src/github.com/algorand/indexer/third_party/go-algorand/ledger/catchupaccessor.go:463","file":"dbutil.go","function":"github.com/algorand/go-algorand/util/db.(*Accessor).atomic","level":"warning","line":304,"msg":"dbatomic: tx surpassed expected deadline by 32.337808794s","readonly":false,"time":"2022-08-02T18:03:01Z"}
Aug 02 18:03:01 ip-10-0-1-62 algorand-indexer[28111]: {"file":"catchpointService.go","function":"github.com/algorand/go-algorand/catchup.(*CatchpointCatchupService).processStageLedgerDownload","level":"warning","line":324,"msg":"unable to download ledger : database or disk is full","time":"2022-08-02T18:03:01Z"}
Aug 02 18:03:03 ip-10-0-1-62 algorand-indexer[28111]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14263996 / 14263996","time":"2022-08-02T18:03:03Z"}
Aug 02 18:03:08 ip-10-0-1-62 algorand-indexer[28111]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14263996 / 14263996","time":"2022-08-02T18:03:08Z"}
Aug 02 18:03:13 ip-10-0-1-62 algorand-indexer[28111]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14263996 / 14263996","time":"2022-08-02T18:03:13Z"}
j1cfr5bghntvy6000000^[Aug 02 18:03:18 ip-10-0-1-62 algorand-indexer[28111]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14263996 / 14263996","time":"2022-08-02T18:03:18Z"}
```

Everything goes well until it doesn't. I was also monitoring my Write and Read OPS in CloudWatch and manually calculating my IOPS which never exceeded 5000 at any given point. Definitely does not feel like fluke so I'm going to try bumping it up to 16000 IOPS and 1000MB throughput and see what happens.

On a side note, this was definitely not an network issue as my betanet and testnet nodes were able to migrate seamlessly with the catchpoints with minimal downtime.

**Comment by phil-vuollet-anterdit** [comment]
similar issue with the new indexer. Here are logs we are seeing:

`{"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 553984 / 14263996","time":"2022-08-02T19:33:49Z"}`

`{"callee":"github.com/algorand/go-algorand/ledger.(*CatchpointCatchupAccessorImpl).processStagingBalances.func3.1","caller":"/go/src/github.com/algorand/indexer/third_party/go-algorand/ledger/catchupaccessor.go:419","file":"dbutil.go","function":"github.com/algorand/go-algorand/util/db.(*Accessor).atomic","level":"warning","line":304,"msg":"dbatomic: tx surpassed expected deadline by 6.944208007s","readonly":false,"time":"2022-08-02T19:33:51Z"}`

**Comment by Blackglade** [comment]
Posting another update. I've maxed out my IOPS (16k) and throughput (1000mb/s) and even switched over from gp3 to io2. Same issues. I get a not enough space and the entire thing resets.

Some additional logs:

```
g 03 02:32:27 ip-10-0-1-62 algorand-indexer[30832]: {"level":"info","msg":"catchup phase 1 of 4 (Processed Accounts): 14267884 / 14267884","time":"2022-08-03T02:32:27Z"}
Aug 03 02:32:27 ip-10-0-1-62 algorand-indexer[30832]: {"callee":"github.com/algorand/go-algorand/ledger.(*CatchpointCatchupAccessorImpl).ResetStagingBalances.func1","caller":"/go/src/github.com/algorand/indexer/third_party/go-algorand/ledger/catchupaccessor.go:191","file":"dbutil.go","function":"github.com/algorand/go-algorand/util/db.(*Accessor).atomic","level":"warning","line":304,"msg":"dbatomic: tx surpassed expected deadline by 8m25.85424858s","readonly":false,"time":"2022-08-03T02:32:27Z"}
Aug 03 02:32:27 ip-10-0-1-62 algorand-indexer[30832]: {"file":"catchpointService.go","function":"github.com/algorand/go-algorand/catchup.(*CatchpointCatchupService).processStageLedgerDownload","level":"warning","line":324,"msg":"unable to download ledger : Get \"http://r-ne.algorand-mainnet.network:4160/v1/mainnet-v1.0/ledger/dg6k0\": dial tcp: lookup r-ne.algorand-mainnet.network on 10.0.0.2:53: no such host","time":"2022-08-03T02:32:27Z"}
```

Downgrading again. It's clear that switching disks or even adjusting the settings isn't making a difference. Btw this is a Mainnet Archival Node if that wasn't clear.

**Comment by winder** [comment]
@phil-vuollet-anterdit  the "context cancelled" error suggests that the process was shutdown. When you start it again it would start the init process from the beginning.

**Comment by phil-vuollet-anterdit** [comment]
> @phil-vuollet-anterdit the "context cancelled" error suggests that the process was shutdown. When you start it again it would start the init process from the beginning.

Had low IOPS increased and got it running. Other issues besides. Will open a new issue.

**Comment by Blackglade** [comment]
Closing out this issue as the problem has been solved. The root cause was indeed a ledger running out of space issue as referenced by this log print:

```
Jul 29 21:57:45 ip-10-0-1-62 algorand-indexer[11463]: {"file":"catchpointService.go","function":"github.com/algorand/go-algorand/catchup.(*CatchpointCatchupService).processStageLedgerDownload","level":"warning","line":324,"msg":"unable to download ledger : database or disk is full","time":"2022-07-29T21:57:45Z"}
```

The core issue is the data directory my indexer service points to is an external separately attached EBS volume with about 1TB of free disk space. My root volume however had only 2GB's available at the time. During phase 1 of the account processing, I was under the assumption that all operations would happen inside the data directory, but the account processing was actually storing the ledger data on my root drive. Once I cleared up space on the root volume and brought it up to 7GB's, the catchup proceeded without issue.
